### PR TITLE
[Heartbeat] Refactor browser summary as a job wrapper

### LINKED ISSUE #33374: [Heartbeat] Monitor reports as summary up: 1 when monitor.status is down and 0 steps ran
I encountered an issue where my monitors were reported as up, despite being down. The resulting heartbeat docs didn't not seem to agree with each other. 

Exampe document
```
{
        "_index": ".ds-synthetics-browser-default-2022.10.17-000001",
        "_id": "klOc54MBl6I9fYHeUTuM",
        "_score": 0,
        "_source": {
          "summary": {
            "up": 1,
            "down": 0
          },
          "agent": {
            "name": "cron-73d66e3e6f7e023e-27767302-vqpvx",
            "id": "376c98ad-ab84-41f9-b019-fbe3d1583ec2",
            "type": "heartbeat",
            "ephemeral_id": "4840234c-936b-4dd6-8e23-1b2eb92195d9",
            "version": "8.4.0"
          },
          "synthetics": {
            "type": "heartbeat/summary"
          },
          "monitor": {
            "origin": "project",
            "name": "staging-8.4.2-ICMP-Create-monitor",
            "project": {
              "name": "synthetics",
              "id": "synthetics"
            },
            "id": "staging-8.4.2-ICMP-Create-monitor-synthetics-default",
            "check_group": "5c0e0386-4e59-11ed-a5e6-2691f1ce5b5b",
            "timespan": {
              "lt": "2022-10-17T20:32:03.629Z",
              "gte": "2022-10-17T20:22:03.629Z"
            },
            "type": "browser",
            "status": "down"
          },
          "error": {
            "type": "io",
            "message": "journey did not finish executing, 0 steps ran: %!w(<nil>)"
          },
          "url": {},
          "tags": [
            "feature",
            "cloud"
          ],
          "observer": {
            "geo": {
              "continent_name": "North America",
              "city_name": "Oregon",
              "country_iso_code": "US",
              "name": "North America - US West",
              "location": "44.000000, -120.500000"
            },
            "hostname": "cron-73d66e3e6f7e023e-27767302-vqpvx",
            "ip": [
              "10.2.0.10"
            ],
            "mac": [
              "26:91:f1:ce:5b:5b"
            ]
          },
          "@timestamp": "2022-10-17T20:22:03.629Z",
          "ecs": {
            "version": "8.0.0"
          },
          "config_id": "c1701c10-4e55-11ed-b398-6faa120a5004",
          "data_stream": {
            "namespace": "default",
            "type": "synthetics",
            "dataset": "browser"
          },
          "event": {
            "agent_id_status": "auth_metadata_missing",
            "ingested": "2022-10-17T20:22:04Z",
            "type": "heartbeat/summary",
            "dataset": "browser"
          }
        }
      },
```
This monitor was generated from the internal synthetics stack e2e repo. It's probably best to attempt to recreate with those monitors. Feel free to reach out to me for more info.

### PR DESCRIPTION
**Pull request by emilioalvap**
## What does this PR do?

Fixes #33374.

Refactors browser summary to be a job wrapper so it can react to job errors.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: heartbeat/monitors/wrappers/wrappers.go [hunk 1]
```diff
@@ -381,3 +377,47 @@ func makeAddSummary() jobs.JobWrapper {
...
+			}
+
+			status, err := event.GetValue("monitor.status")
+			if err != nil {
+				return nil, fmt.Errorf("could not wrap summary for '%s', no status assigned: %w", sf.ID, err)
+			}
+
+			up, down := 1, 0
+			if monitorstate.StateStatus(status.(string)) == monitorstate.StatusDown {
+				down, up = up, down
```
Comment (andrewvc) [inline heartbeat/monitors/wrappers/wrappers.go:399]:
```suggestion
				up, down = 0, 1
```

This is more clear, and in terms of performance, either negligible or zero due to compiler optimizations

**Review by andrewvc — APPROVED** [review]
LGTM pending the one change

**Comment by justinkambic** [comment]
Post-FF testing LGTM

**Pull request merged**